### PR TITLE
Rework watchlist top button visibility, Allow student names to be copied

### LINKED ISSUE #1612: Not possible to copy student names from watchlist screen
**Your environment**
 Carnegie Mellon's Autolab at https://autolab.andrew.cmu.edu/

**Steps To Reproduce**

1. Go to watchlist screen (https://[AUTOLAB INSTANCE]/courses/[COURSE]/metrics/index)
2. Select (using browser's built-in text selection mechanism) one or more rows of the table.
3. Copy text.
4. Paste into a text editor.

**Current behavior**

The only thing copied from the selected rows is the text of the tags explaining why each student was flagged for attention.  The students' names and email addresses are not copied.

**Expected behavior**

Students' names and email addresses should be copied.

**Additional**

This appears to be caused by this CSS rule:

```
.ui.checkbox input.hidden+label {
 cursor:pointer;
 -webkit-user-select:none;
 -moz-user-select:none;
 -ms-user-select:none;
 user-select:none
}
```

It should be more carefully targeted.  As a general rule, text that presents _data_ should always be selectable and copiable.  I suspect this rule was _intended_ to apply to text, elsewhere in the application, that presents controls and is appropriately not selectable (like the "contact" and "resolve" buttons in this table, which are _correctly_ not copied) and then the styling got reused for this and nobody checked copy and paste behavior.

### PR DESCRIPTION
**Pull request by 20wildmanj**
## Description

As part of UI Audit. 

- Previously, select all for watchlist had no tooltip / description. Also the checkbox is not centered with the other buttons. 

Screen Shot 2022-10-23 at 16 51 56

- The top buttons would also be disabled in watchlist tabs where they should be hidden, because there is no use-case where they would be enabled.  The select all checkbox also would be rendered as well.
  - contact button in contacted tab never gets used
  - contact and resolve in resolved tab never gets used
  - contact and resolve in archived tab similarly

Current Contacted tab: (note how the student-specific actions doesn't have the contact option)
Screen Shot 2022-10-23 at 16 14 36 

Current Pending with no students on pending:
Screen Shot 2022-10-23 at 16 14 50

Current archived (contact and resolved disabled, but delete allowed):
Screen Shot 2022-10-23 at 16 14 57

Current Resolved (no options available, but still shows up as disabled):
Screen Shot 2022-10-23 at 16 54 35

## Changes

- Move top buttons into a flexbox to help center checkbox, add text label to select all checkbox
- Add `showTopButtons` to `watchlist.js` helper function to hide and show the top buttons based on watchlist tab, instead of disabling them, update `updateButtonVisibility` to use new function
- Add some slight documentation to `watchlist.js`
- Also change css for student single select checkbox to  allow student names to be copied from watchlist screen

Closes #1612

New top tab:
Screen Shot 2022-10-23 at 16 58 59

Pending with no students:
Screen Shot 2022-10-23 at 17 00 58

Contacted with / without students:
Screen Shot 2022-10-23 at 17 00 34
Screen Shot 2022-10-23 at 17 01 04

Resolved:
Screen Shot 2022-10-23 at 17 00 40

Archived:
Screen Shot 2022-10-23 at 17 00 49

## How Has This Been Tested?

- View watchlist with students on pending, see all options except for delete appear
- View watchlist with students on contacted, see only select all and resolve appear
- View watchlist with no students on pending/contacted, see no options appear
- View resolved watchlist tab, see no options appear
- View archived watchlist tab, see only select all and delete appear
- Copy student name, email from watchlist

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run rubocop for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting

### CODE REVIEW

**Comment by damianhxy** [comment]
Nit:  Students on the resolved watchlist tab have a checkbox next to their names, even though there are no options. We should consider hiding the checkboxes as well.

Screen Shot 2022-10-24 at 18 33 42

File: app/assets/javascripts/watchlist.js [hunk 1]
```diff
@@ -547,35 +549,58 @@ $('.ui.vertical.fluid.tabular.menu .item').on('click', function() {
   updateButtonVisibility(this);
 });
 
+
+
+// helper function to configure top button visibility
+function showTopButtons(selectAllCheckbox, resolveButton, contactButton, deleteButton) {
+  if (selectAllCheckbox === true) {
```
[the remark this answers is not in the document]
Comment (damianhxy) [reply app/assets/javascripts/watchlist.js:556]: https://api.jquery.com/toggle/#toggle-display

**Review by damianhxy — APPROVED** [review]
Followed testing plan, interface works as expected. Text can also now be selected.

Left some comments, but otherwise lgtm!

File: app/assets/javascripts/watchlist.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (20wildmanj) [reply app/assets/javascripts/watchlist.js]: nifty!

**Comment by 20wildmanj** [comment]
> Nit: Students on the resolved watchlist tab have a checkbox next to their names, even though there are no options. We should consider hiding the checkboxes as well.
> 
> Screen Shot 2022-10-24 at 18 33 42

Done, for some reason checkbox classes apply different styles to the label so the name and email look a tiny bit different from the ones with a checkbox.
Screen Shot 2022-10-25 at 23 07 53
Screen Shot 2022-10-25 at 23 16 49

**Comment by damianhxy** [comment]
Changes look good, however the difference in style is slightly jarring.

Screen Shot 2022-10-27 at 00 13 58

This seems to be the style applied by the checkbox, specifically via the `[type="checkbox"]+label` selector. Perhaps we could apply the style to all labels by default? Alternatively, we could overwrite this style (it makes the text harder to read imo)

**Comment by 20wildmanj** [comment]
> Changes look good, however the difference in style is slightly jarring.
> 
> Screen Shot 2022-10-27 at 00 13 58
> 
> This seems to be the style applied by the checkbox, specifically via the `[type="checkbox"]+label` selector. Perhaps we could apply the style to all labels by default? Alternatively, we could overwrite this style (it makes the text harder to read imo)

I found a way to override, but going to investigate a bit more because this is firefox:
Screen Shot 2022-10-27 at 12 19 24

this is chrome:
Screen Shot 2022-10-27 at 12 19 37

I don't know why the style just looks a lot uglier in other browsers so will try to tune this up

**Comment by 20wildmanj** [comment]
Ok @damianhxy should be more consistent now:

Chrome:
Screen Shot 2022-10-27 at 12 43 45
Screen Shot 2022-10-27 at 12 43 51

Firefox:
Screen Shot 2022-10-27 at 12 43 59
Screen Shot 2022-10-27 at 12 44 05

**Comment by damianhxy** [comment]
Tested in both Firefox and Chrome.

LGTM!

**Pull request merged**